### PR TITLE
[LLAPI] add setLang and setScore

### PR DESCRIPTION
**Pull request by ashtul**
1. Moved language enum and functions to `language.c/.h`
2. Removed `stemmer.h` from `redisearch_api.h` and added `language.h` to `redisearch_api.c`.
3. Added `RediSearch_IndexOptionsSetScore`, `RediSearch_IndexOptionsSetLanguage` and `RediSearch_ValidateLanguage`.

### CODE REVIEW

File: src/language.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#ifndef __RS_LANGUAGE_H__
+#define __RS_LANGUAGE_H__
```
Comment (rafie) [inline src/language.h:2]: it's 2022 very soon, can do `#pragma once`
:-)

**Comment by rafie** [comment]
@ashtul So we decided to go with the string option, not the enum one?

**Comment by ashtul** [comment]
@rafie yes.
A `RediSearch_ValidateLanguage` was added to bridge the gap.

File: src/language.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  { "spanish",    RS_LANG_SPANISH },
+  { "swedish",    RS_LANG_SWEDISH },
+  { "tamil",      RS_LANG_TAMIL },
+  { "turkish",    RS_LANG_TURKISH },
+  { "yiddish",    RS_LANG_YIDDISH },
+  { "chinese",    RS_LANG_CHINESE },
+  { NULL,         RS_LANG_UNSUPPORTED }
+};
+
+const char *RSLanguage_ToString(RSLanguage language) {
```
Comment (oshadmi) [inline src/language.c:43]: Can we use `language` as the index in `__langPairs` (checking it is not beyond `RS_LANG_UNSUPPORTED`)
And returning the `str` in this slot?

**Review by rafie — APPROVED** [review]

**Comment by rafie** [comment]
@ashtul Let's CP it to 2.2.

**Pull request merged**